mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -49,6 +49,20 @@ private[feature] trait QuantileDiscretizerBase extends Params
 
   /** @group getParam */
   def getNumBuckets: Int = getOrDefault(numBuckets)
+
+  /**
+   * Relative error (see approxQuantile documentation for description).  Must be >= 0.
+   * default: 0.01
+   * @group param
+   */
+  val relativeError = new DoubleParam(this, "relativeError", "The relative target precision " +
+    "for approxQuantile",
+    ParamValidators.gtEq(0.0))
+  setDefault(relativeError -> 0.01)
```

Comment:
If we go for the formula as the default, I think it's misleading to have a `setDefault` here which is not actually used, and the comment also indicates the default is `0.01` which it isn't actually.